homeassistant/components/advantage_air/climate.py

```diff
@@ -92,37 +93,37 @@ class AdvantageAirAC(AdvantageAirAcEntity, ClimateEntity):
         ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
     )
 
-    def __init__(self, instance, ac_key):
+    def __init__(self, instance, ac_key: str) -> None:
         """Initialize an AdvantageAir AC unit."""
         super().__init__(instance, ac_key)
         if self._ac.get("myAutoModeEnabled"):
             self._attr_hvac_modes = AC_HVAC_MODES + [HVACMode.AUTO]
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float:
         """Return the current target temperature."""
         return self._ac["setTemp"]
 
     @property
-    def hvac_mode(self):
+    def hvac_mode(self) -> HVACMode | str | None:
```

Comment:
ADVANTAGE_AIR_HVAC_MODES only contains `HVACMode` values, so you can drop `str` here.
```suggestion
    def hvac_mode(self) -> HVACMode | None:
```